bpf/aya-bpf/src/btf_args.rs

```diff
@@ -0,0 +1,46 @@
+use crate::cty::c_void;
+
+/// A trait that indicates a valid type for an argument which can be coerced from a BTF
+/// context.
+///
+/// Users should not implement this trait.
+///
+/// SAFETY: This trait is _only_ safe to implement on primitive types that can fit into
+/// a `u64`. For example, integers and raw pointers may be coerced from a BTF context.
+pub unsafe trait FromBtfArgument {
+    /// Coerces a `T` from the `n`th argument from a BTF context where `n` starts
+    /// at 0 and increases by 1 for each successive argument.
+    ///
+    /// SAFETY: This function is deeply unsafe, as we are reading raw pointers into kernel memory.
+    /// In particular, the value of `n` must not exceed the number of function arguments.
+    /// Luckily, the BPF verifier will catch this for us.
+    unsafe fn from_argument(ctx: *const c_void, position: usize) -> Self;
+}
+
+unsafe impl<T> FromBtfArgument for *const T {
+    unsafe fn from_argument(ctx: *const c_void, position: usize) -> *const T {
+        *(ctx as *const *const c_void).add(position * 8) as _
```

Comment:
I'm not sure I understand, shouldn't it be `ctx.add((1 + position) * mem::size_of::<*const c_void>()) as _`?

Reply:
Good catch, I'll fix it

Reply:
@alessandrod I've updated this to be more semantically meaningful. While I was at it I just switched to use ptr.offset instead of ptr.add. Much simpler and more correct.